modules/node/persistence/cms_node_storage_sql.e

```diff
@@ -255,6 +255,32 @@ feature -- Access
 --			end
 		end
 
+
+	available_parents_for_node (a_node: CMS_NODE): LIST [CMS_NODE]
```

Comment:
I think there is no check for cycle.
Maybe this is ensured by component in charge to edit and store the page.
In this case, we need a comment and a contract to ensure no cycle exists.
Otherwise, the code should be updated to remove any potential cycle issue.